### PR TITLE
Stricter

### PR DESCRIPTION
**Pull request by grempe**
Each commit should be very small and self-contained to make it easier to review. These are mostly tiny fixes for jshint and de-crufting bits of old code I found.  Getting to the bottom of these now. This pass is just addressing issues in the server dir.

Perhaps you can take a closer look at the FIXME statements I added as well.  I didn't want to address those in detail on this quick pass.

### CODE REVIEW

File: server/lib/account.js [hunk 1]
```diff
@@ -199,7 +202,10 @@ Account.prototype.update = function () {
   // update({ key: 'value' });
   if (typeof arguments[0] === 'object') {
     for (var key in arguments[0]) {
-      this[key] = arguments[0][key];
+        if (arguments[0].hasOwnProperty(key)) {
```
Comment (daviddahl) [inline server/lib/account.js]: :+1:  oh yeah.

File: server/lib/account.js [hunk 2]
```diff
@@ -333,6 +340,7 @@ Account.prototype.sendMessage = function (options, callback) {
  * @param {Function} callback
  */
 Account.prototype.changePassphrase = function (keyring, callback) {
+  // FIXME ; jshint : confusing use of !
   if (!keyring && !(typeof keyring == 'object')) {
```
Comment (daviddahl) [inline server/lib/account.js]: ah. weird. should just be:
`if (typeof keyring !== 'object')`

Comment (grempe) [inline server/lib/account.js]: I'll change it.

File: server/lib/transaction.js [hunk 3]
```diff
@@ -104,7 +104,9 @@ Transaction.prototype.update = function () {
   // update({ key: 'value' });
   if (typeof arguments[0] === 'object') {
     for (var key in arguments[0]) {
```
Comment (daviddahl) [inline server/lib/transaction.js]: I'm thinking all of these should eventually just be `for(var i = 0; i < foo.length; i++)`

Comment (daviddahl) [inline server/lib/transaction.js]: ... as these have caused some massive debugging headaches in the past

File: server/lib/account.js (hunk 2 above)
Comment (grempe) [inline server/lib/account.js]: Pushed change to this branch

File: server/routes/account.js [hunk 4]
```diff
@@ -151,23 +155,26 @@ app.post('/account/:username/answer', function (req, res) {
...
-	    return;
-	  }
-	  res.send({
-            success: true,
-            account: account.toJSON(),
-            sid: sid,
-            srpM2: srpM2.toString('hex')
-           });
-	});
+        // add accountId to session.
```
Comment (daviddahl) [inline server/routes/account.js]: this reminds me of the fact that we temp store SRP hash in redis, but need to remove it upon finally logging in.

File: server/routes/item.js [hunk 5]
```diff
@@ -19,7 +19,6 @@
 'use strict';
 
 var app = process.app;
-var db = app.datastore;
```
Comment (daviddahl) [inline server/routes/item.js]: whoops. these were left over after doing some experiments! Great catch!

Comment (grempe) [inline server/routes/item.js]: That's our friend jshint you can thank.  I am also looking at http://jscs.info

I think both of these should be made to run on test runs and failures should fail test.

**Comment by daviddahl** [comment]
:+1:  WFM +1 LGTM, FUCK YEAH

File: server/routes/item.js (hunk 5 above)
Comment (daviddahl) [inline server/routes/item.js]: yeah, this is very cool. Is there an npm module for making all of this run - or are there non-node dependencies?

Comment (grempe) [inline server/routes/item.js]: Both jshint and jscs are just npm modules to install globally. No other depends.  So if we switch to something like Grunt/Gulp or its ilk we can define them to run on test runs.  I just installed jscs so I'll see how painful its reports are when running on current code.  :-)

Comment (daviddahl) [inline server/routes/item.js]: This is a great plan, not sure if we can use Grunt/Gulp for just this while a Makefile excise PR awaits?

Comment (grempe) [inline server/routes/item.js]: I see no reason the two cannot co-exist.  Start building up Grunt/Gulp (better suggestion?) and remove parts from Makefile along the way.

Comment (daviddahl) [inline server/routes/item.js]: :+1:

**Comment by devgeeks** [comment]
We could also just use a locally installed jshint and `npm rum` style scripting until we settle on a build tool.

**Comment by devgeeks** [comment]
*sigh* `npm run`

Stooped phone.

**Pull request merged**